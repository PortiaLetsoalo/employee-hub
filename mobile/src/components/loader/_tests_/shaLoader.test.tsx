import React from 'react';
import { create, act } from 'react-test-renderer';
import ShaLoader from '../shaLoader';

jest.mock('react-native-paper', () => ({
    ActivityIndicator: 'ActivityIndicator',
}));

const tree = create(<ShaLoader loading></ShaLoader>);
test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});

test('shaLoader isDefined', () => {
  expect(<ShaLoader />).toBeDefined();
});

test('shaLoader render', () => {
  expect(tree.root.props.loading).toEqual(true);
  act(() => {
    tree.update(<ShaLoader loading type="overlay"></ShaLoader>);
  });
  expect(tree.root.props.type).toEqual('overlay');
});
