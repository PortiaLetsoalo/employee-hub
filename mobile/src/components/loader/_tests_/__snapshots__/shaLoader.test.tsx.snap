// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    size="large"
  />
</View>
`;
